### DISCUSSION TITLE
Extra sorting options for kata in user profile

### DISCUSSION THREAD

**Discussion by Captain-Quack** [codewars/codewars.com#2918] [Ideas]
(*this is a standalone idea from [this](https://github.com/codewars/codewars.com/discussions/2888) list*)
It would be nice to be able to sort the list of completed/unfinished kata by the same filters as the browser (Oldest, Hardest, Popularity...)
It could also introduce new filters, like solution length, attempts made, or the number of languages solved.

**Comment by Kacarott** [comment]
As an alternative to adding sorting/filters to the "unfinished kata" section of the profile, a filter "Kata I have started but not finished" could be added to the "Progress" filter in the main kata page.

**Reply by hobovsky** [reply]
Isn't this the "Kata I have not completed" filter option?

![image](https://user-images.githubusercontent.com/23709795/214884031-5bdd08fa-daa9-4075-9169-bef01acd2a9f.png)

**Reply by Kacarott** [reply]
This is every kata you haven't completed, which is different to kata you have started but not completed.

**Reply by hobovsky** [reply]
Oh right, I mixed up things.

Then yes, I think such filter would be nice here. Potentially backed up with a functional improvement that just entering the trainer does not mark kata as trained on until any of TEST or ATTEMPT is pressed at least once.

OTOH, after entering the trainer you can see sample tests, what makes it possible to train outside of the site, so it kind of makes sense to mark such kata as (potentially) trained on. Maybe a kata being trained on is just not a good metric? Maybe something like "attempted, but not passed" would be more suitable?
